### PR TITLE
Token updates

### PR DESCRIPTION
**Pull request by anonymoussprocket**
- Balance transfer fix for @StakerDAO
- update michelson parser to remove DUP n macro

### CODE REVIEW

**Review by vishakh — APPROVED** [review]

File: src/chain/tezos/contracts/DexterPoolHelper.ts [hunk 1]
```diff
@@ -103,6 +103,34 @@ export namespace DexterPoolHelper {
...
+     * @returns Raw values that are not scaled.
+     */
+    export async function getAccountPoolShare(server: string, address: string, account: string): Promise<{token: number, xtz: number}> {
+        try {
+            const storage = await getSimpleStorage(server, address);
+
+            const packedKey = TezosMessageUtils.encodeBigMapKey(Buffer.from(TezosMessageUtils.writePackedData(account, 'address'), 'hex'));
+            const mapResult = await TezosNodeReader.getValueForBigMapKey(server, storage.balanceMap, packedKey);
+
+            if (mapResult === undefined) { throw new Error(`Map ${storage.balanceMap} does not contain a record for ${account}`); }
```
Comment (bhaskarkishore) [inline src/chain/tezos/contracts/DexterPoolHelper.ts:120]: We are throwing errors to short circuit a try/catch block but the error is not consumed in any way. I assume the motivation here was to have fixed return paths but this looks odd. Perhaps we can log the error in the `catch` to make it somewhat meaningful?

**Review by bhaskarkishore — APPROVED** [review]
Reviewed as best as possible.

File: src/chain/tezos/contracts/DexterPoolHelper.ts (hunk 1 above)
Comment (anonymoussprocket) [reply src/chain/tezos/contracts/DexterPoolHelper.ts:120]: This code is presently not in use.

Comment (bhaskarkishore) [reply src/chain/tezos/contracts/DexterPoolHelper.ts:120]: Regardless, this is not a good pattern. We should avoid it if possible.

**Pull request merged**